pennylane/templates/subroutines/uccsd.py

```diff
@@ -118,31 +117,34 @@ def UCCSD(weights, wires, ph=None, pphh=None, init_state=None):
 
             import pennylane as qml
             from pennylane.templates import UCCSD
+            from functools import partial
 
-            # Number of active electrons
-            n_active_electrons = 2
+            # Build the electronic Hamiltonian
+            name = "h2"
+            geo_file = "h2.xyz"
+            h, qubits = qml.qchem.molecular_hamiltonian(name, geo_file)
 
-            # Number of active spin-orbitals
-            n_active_orbitals = 4
+            # Number of electrons
+            electrons = 2
 
-            # Set the ``ref_state`` array to encode the HF state
-            ref_state = qml.qchem.hf_state(n_active_electrons, n_active_orbitals)
+            # Define the HF state
+            ref_state = qml.qchem.hf_state(electrons, qubits)
 
-            # Spin-projection selection rule to generate the excitations
-            DSz = 0
+            # Generate single and double excitations
+            singles, doubles = qml.qchem.excitations(electrons, qubits)
 
-            # Generate the particle-hole configurations
-            ph, pphh = qml.qchem.sd_excitations(n_active_electrons, n_active_orbitals, DS=DSz)
+            # Map excitations to the wires the UCCSD circuit will act on
+            s_wires, d_wires = qml.qchem.excitations_to_wires(singles, doubles)
 
-            dev = qml.device('default.qubit', wires=n_active_orbitals)
+            # Define the device
+            dev = qml.device('default.qubit', wires=qubits)
 
-            @qml.qnode(dev)
-            def circuit(weights, wires, ph=ph, pphh=pphh, init_state=ref_state):
-                UCCSD(weights, wires, ph=ph, pphh=pphh, init_state=ref_state)
-                return qml.expval(qml.PauliZ(0))
+            # Define the UCCSD ansatz
+            ansatz = partial(UCCSD, init_state=ref_state, s_wires=s_wires, d_wires=d_wires)
 
-            params = np.random.normal(0, np.pi, len(ph) + len(pphh))
-            print(circuit(params, wires, ph=ph, pphh=pphh, init_state=ref_state))
+            # Compute the expectation value of 'h' for given set of parameters 'params'
+            params = np.random.normal(0, np.pi, len(singles) + len(doubles))
+            print(qml.VQECost(ansatz, h, dev))
```

Comment:
Just checking that you have double checked that this example still runs after the changes!

Reply:
Thanks @josh146. In order to check it within this PR I would need to use the old names of the functions that were refactored but have not been merged yet. I can go with the old functions, check and update the names afterwards or update the name of the functions here and double check before merging. I will keep it in mind.